examples/pytorch/CaptumExample/Titanic_Captum_Interpret.py

```diff
@@ -0,0 +1,394 @@
+    Age: Passenger Age
+    Sibsp: Number of Siblings / Spouses Aboard
+    Parch: Number of Parents / Children Aboard
+    Fare: Fare Amount Paid in British Pounds
+    Female: Binary variable indicating whether passenger is female
+    Male: Binary variable indicating whether passenger is male
+    EmbarkC : Binary var indicating whether passenger embarked @ Cherbourg
+    EmbarkQ : Binary var indicating whether passenger embarked @ Queenstown
+    EmbarkS : Binary var indicating whether passenger embarked @ Southampton
+    Class1 : Binary var indicating whether passenger was in first class
+    Class2 : Binary var indicating whether passenger was in second class
+    Class3 : Binary var indicating whether passenger was in third class
+    url = "https://biostat.app.vumc.org/wiki/pub/Main/DataSets/titanic3.csv"
+    """
+    url = "https://biostat.app.vumc.org/wiki/pub/Main/DataSets/titanic3.csv"
+    download_from_url(url, root="data")
+    dataset_path = "data/titanic3.csv"
+    titanic_data = pd.read_csv(dataset_path)
+
+    titanic_data = pd.concat(
+        [
+            titanic_data,
+            pd.get_dummies(titanic_data["sex"]),
+            pd.get_dummies(titanic_data["embarked"], prefix="embark"),
+            pd.get_dummies(titanic_data["pclass"], prefix="class"),
+        ],
+        axis=1,
+    )
+
+    titanic_data["age"] = titanic_data["age"].fillna(titanic_data["age"].mean())
+    titanic_data["fare"] = titanic_data["fare"].fillna(titanic_data["fare"].mean())
+    titanic_data = titanic_data.drop(
+        ["name", "ticket", "cabin", "boat", "body", "home.dest", "sex", "embarked", "pclass",],
+        axis=1,
+    )
+    return titanic_data
+
+
+torch.manual_seed(1)  # Set seed for reproducibility.
+
+
+class TitanicSimpleNNModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear1 = nn.Linear(12, 12)
+        self.sigmoid1 = nn.Sigmoid()
+        self.linear2 = nn.Linear(12, 8)
+        self.sigmoid2 = nn.Sigmoid()
+        self.linear3 = nn.Linear(8, 2)
+        self.softmax = nn.Softmax(dim=1)
+
+    def forward(self, x):
+        lin1_out = self.linear1(x)
+        sigmoid_out1 = self.sigmoid1(lin1_out)
+        sigmoid_out2 = self.sigmoid2(self.linear2(sigmoid_out1))
+        return self.softmax(self.linear3(sigmoid_out2))
+
+
+def prepare():
+    np.random.seed(131254)
+    titanic_data = TitanicDataset()
+    labels = titanic_data["survived"].to_numpy()
+    titanic_data = titanic_data.drop(["survived"], axis=1)
+    feature_names = list(titanic_data.columns)
+    data = titanic_data.to_numpy()
+
+    # Separate training and test sets using
+    train_indices = np.random.choice(len(labels), int(0.7 * len(labels)), replace=False)
+    test_indices = list(set(range(len(labels))) - set(train_indices))
+    train_features = data[train_indices]
+    train_labels = labels[train_indices]
+    test_features = data[test_indices]
+    test_labels = labels[test_indices]
```

Comment:
Can we replace these lines with `train_test_split` in `sklearn`?

Reply:
@harupy  Yes , I have updated this.